sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/expressions/CodeGenerationSuite.scala

```diff
@@ -97,6 +97,38 @@ class CodeGenerationSuite extends SparkFunSuite with ExpressionEvalHelper {
     assert(actual(0) == cases)
   }
 
+  test("SPARK-18091: split large if expressions into blocks due to JVM code size limit") {
+    val row = create_row("afafFAFFsqcategory2dadDADcategory8sasasadscategory24", 0)
```

Comment:
> Since the fix in this pull request is only in If expression, a working testcase with the fix will be one which consists (recursively) of If expressions.

I can't see why this is reasonable. You just need to construct enough big generated codes of condition + true expression + false expression.